assets/js/theme/common/product-details.js

```diff
@@ -25,6 +25,21 @@ export default class ProductDetails extends ProductDetailsBase {
         const hasOptions = $productOptionsElement.html().trim().length;
         const hasDefaultOptions = $productOptionsElement.find('[data-default]').length;
         const $productSwatchGroup = $('[id*="attribute_swatch"]', $form);
+        const $productSwatchLabels = $('.form-option-swatch', $form);
+
+        $(window).on('load', () => {
+            $.each($productSwatchLabels, (_, label) => {
+                const $optionImage = $('.form-option-expanded', $(label));
+                const optionImageWidth = $optionImage.outerWidth();
+                const { right } = label.getBoundingClientRect();
+                const emptySpaceToScreenRightBorder = window.screen.width - right;
+                const shiftValue = optionImageWidth - emptySpaceToScreenRightBorder;
+
+                if (emptySpaceToScreenRightBorder < (optionImageWidth + 55)) {
```

Comment:
Why are we using 55 here?

Reply:
it represents 55px from css styles - currently swatch image has next shift:
`.form-option-expanded {
    left: calc(100% + 55px);
}`
I check if empty space on screen is enough to hold 55px offset alongside scale image width. 

As an option I can add a separate variable that will help to understand this number.